src/main/java/net/snowflake/client/core/SFStatement.java

```diff
@@ -787,7 +791,10 @@ public SFBaseResultSet execute(
       throws SQLException, SFException {
     TelemetryService.getInstance().updateContext(session.getSnowflakeConnectionString());
 
-    // todo: if (dataframeAst == null)
+    if (dataframeAst != null) {
```

Comment:
Instead of return early, just use if statement to skip all sql functions, then we can benefit from feature changes related to non-sql part.